### PR TITLE
:memo: First cut of providers page

### PR DESCRIPTION
**Pull request by rowanmiller**

### CODE REVIEW

**Comment by rowanmiller** [comment]
cc @ErikEJ and @roji

**Comment by roji** [comment]
Great.

* EntityFramework.Npgsql (with a capital N) is better
* The Npgsql site is http://www.npgsql.org
* There's a documentation placeholder page for EF7 at http://www.npgsql.org/doc/ef7.html, I guess it makes sense to link there for getting started...

**Comment by ErikEJ** [comment]
LGTM

**Comment by ErikEJ** [comment]
Maybe info about supported platfroms for each provider would be useful, I have already had someone attempt to use sqlce on dnx...

**Comment by rowanmiller** [comment]
@ErikEJ great point, I will add it in

**Comment by rowanmiller** [comment]
Added a platforms section to each provider.

@roji did you want me to link to npgsql.org instead of GitHub for the "project site"?

**Comment by roji** [comment]
Yeah, I think that would be better...

File: docs/providers/index.rst [hunk 1]
```diff
@@ -1,8 +1,95 @@
-.. include:: /stub-topic.txt
+Database Providers
+==================
 
-|stub-icon| Providers
-=====================
+The following providers are either available or being developed:
+ - `EntityFramework.SqlServer`_
+ - `EntityFramework.SQLite`_
```
Comment (bricelam) [inline docs/providers/index.rst]: "Sqlite", not "SQlite"

**Pull request merged**